ghmirror/core/mirror_requests.py

```diff
@@ -83,10 +108,28 @@ def online_request(method, url, auth, data=None):
         if last_mod is not None:
             headers['If-Modified-Since'] = last_mod
 
-    resp = requests.request(method=method, url=url, headers=headers,
-                            timeout=REQUESTS_TIMEOUT)
+    resp = requests.request(method=method,
+                            url=url,
+                            headers=headers,
+                            timeout=REQUESTS_TIMEOUT,
+                            params=parameters)
 
     if resp.status_code == 304:
+        if len(cached_response.json()) == per_page_elements and \
+           cached_response.links == {}:
+
+            headers.pop('If-None-Match')
+            resp = requests.request(method=method,
+                                    url=url,
+                                    headers=headers,
+                                    timeout=REQUESTS_TIMEOUT,
+                                    params=parameters)
+
+            LOG.info('ONLINE GET CACHE_MISS %s', url)
+            cache[cache_key] = resp
```

Comment:
we do not cache ANY response... it has to be a 200 and it has to contain the conditional requests headers so we cache it:

https://github.com/app-sre/github-mirror/blob/master/ghmirror/core/mirror_requests.py#L108-L109

Maybe we should de-duplicate that routine?